### PR TITLE
chore(lua): add vscode and luacheck setup

### PR DESCRIPTION
**Pull request by dtfiedler**

### CODE REVIEW

File: src/common/main.lua [hunk 1]
```diff
@@ -515,7 +515,7 @@ function ant.init()
 			end
 
 			local srcCodeTxIdStatus, srcCodeTxIdResult = pcall(utils.validateArweaveId, srcCodeTxId)
-			if srcCodeTxIdStatus and not srcCodeTxIdStatus then
+			if srcCodeTxIdStatus and not srcCodeTxIdResult then
```
Comment (dtfiedler) [inline src/common/main.lua:518]: this was caught by the linter - so I modified to what I thought made sense. now it's failing evolve tests.

@atticusofsparta - can you take a look?

Comment (dtfiedler) [reply src/common/main.lua]: looks like this should be [7b54a93](https://github.com/ar-io/ar-io-ant-process/pull/29/commits/7b54a93cfb1b220404b16027086e8cda58415511)

File: src/common/json.lua [hunk 2]
```diff
@@ -49,7 +49,7 @@ local function escape_char(c)
 	return "\\" .. (escape_char_map[c] or string.format("u%04x", c:byte()))
 end
 
-local function encode_nil(val)
+local function encode_nil()
```
Comment (atticusofsparta) [inline src/common/json.lua:52]: considering reverting these and making a pr to json.lua repo - maybe we should maintain our own fork though.

**Review by atticusofsparta — APPROVED** [review]

File: src/common/json.lua (hunk 2 above)
Comment (dtfiedler) [reply src/common/json.lua:52]: it's just a copied file - don't think we need to fork

**Pull request merged**